### ISSUE TITLE
Add error message to maven-install.sh which doesn't work with Maven 3.9.0 due to MNG-7679

### ISSUE THREAD

**Issue by CyborTronik** [jOOQ/jOOQ#15072]
### Expected behavior

Maven scripts should pass succesfull

### Actual behavior

Got an error
```
$ ./maven-install.sh 
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------< org.apache.maven:standalone-pom >-------------------
[INFO] Building Maven Stub Project (No POM) 1
[INFO] --------------------------------[ pom ]---------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.070 s
[INFO] Finished at: 2023-05-15T12:24:17+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Cannot invoke "java.util.List.stream()" because the return value of "org.apache.maven.project.MavenProject.getCollectedProjects()" is null
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.

```

### Steps to reproduce the problem

download express edition of jooq. unzip it. from root of the project mark executable and run `./maven-install.sh`

### jOOQ Version

3.18.4

### Database product and version

Postgres 15

### Java Version

18.0.1

### OS Version

Debian GNU/Linux 11

**Comment by lukaseder** [comment]
I understand you've preferred a refund over troubleshooting this issue, so I'm closing it as "worksforme":
https://github.com/jOOQ/jOOQ/issues/9906#issuecomment-1547662960

Let me know if I can be of any further help, nonetheless.

**Comment by CyborTronik** [comment]
Why is closed? Was it solved? 
Was it about money or experience?

**Comment by lukaseder** [comment]
@CyborTronik You've asked for a refund, which I've already executed, so I take this issue won't need further investigation.

My assumption here was that you might have used an outdated maven version? If you'd still like this issue addressed, can you please provide some additional information on how to reproduce this?

**Comment by CyborTronik** [comment]
Is this outdated:
```
$ mvn --version
Apache Maven 3.9.0 (9b58d2bad23a66be161c4664ef21ce219c2c8584)
Maven home: /home/stanislav/.sdkman/candidates/maven/current
Java version: 18.0.1, vendor: Azul Systems, Inc., runtime: /home/stanislav/.sdkman/candidates/java/18.0.1-zulu
Default locale: pt_PT, platform encoding: UTF-8
OS name: "linux", version: "5.10.0-23-amd64", arch: "amd64", family: "unix"

```

**Comment by lukaseder** [comment]
It doesn't look like it. Can you add the `-e` flag and show the complete stack trace?

**Comment by lukaseder** [comment]
Googling the error message leads to this issue: https://github.com/jeremylong/DependencyCheck/issues/5437

Which indeed leads to a Maven bug, showing that 3.9.0 had a problem:
https://issues.apache.org/jira/browse/MNG-7679

It might go away when you upgrade to 3.9.1

**Comment by lukaseder** [comment]
Another reference to the same bug:
https://github.com/liquibase/liquibase/pull/3828

**Comment by lukaseder** [comment]
I'll investigate this to see what can be done. At the least, a pointer to the maven bug would be helpful to help users troubleshoot this problem. There might be others running into this.

**Comment by lukaseder** [comment]
I can reproduce the issue with Maven 3.9.0:

```
$ ./maven-install.sh
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8 -Duser.language=en
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------< org.apache.maven:standalone-pom >-------------------
[INFO] Building Maven Stub Project (No POM) 1
[INFO] --------------------------------[ pom ]---------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.149 s
[INFO] Finished at: 2023-05-16T09:30:12+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Cannot invoke "java.util.List.stream()" because the return value of "org.apache.maven.project.MavenProject.getCollectedProjects()" is null
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.

lukas@DESKTOP-F2SE3C1 MINGW64 ~/Downloads/jOOQ-3.18.4
$ mvn -version
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8 -Duser.language=en
Apache Maven 3.9.0 (9b58d2bad23a66be161c4664ef21ce219c2c8584)
Maven home: C:\apache-maven-3.9.0
Java version: 19.0.1, vendor: Eclipse Adoptium, runtime: C:\Program Files\Java\jdk-19
Default locale: en_CH, platform encoding: UTF8
OS name: "windows 11", version: "10.0", arch: "amd64", family: "windows"
```

**Comment by lukaseder** [comment]
I'll add these messages to the batch script:

```
for /F "delims=" %%i in ('mvn -version ^| findstr /C:"Apache Maven 3.9.0"') do (
  echo.
  echo ERROR: Maven 3.9.0 has a known bug preventing install:install-file calls.
  echo Please upgrade to a newer version.
  echo See also: https://issues.apache.org/jira/browse/MNG-7679
  echo.
)
```

And bash script:

```
BAD_MAVEN_VERSION=".*Apache Maven 3.9.0.*"
MAVEN_VERSION=$(mvn -version | grep '.*Apache Maven 3\.9\.0.*')
if [[ "$MAVEN_VERSION" =~ $BAD_MAVEN_VERSION ]]; then
  echo ""
  echo "ERROR: Maven 3.9.0 has a known bug preventing install:install-file calls."
  echo "Please upgrade to a newer version."
  echo "See also: https://issues.apache.org/jira/browse/MNG-7679"
  echo ""
fi
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.19.0, 3.18.5 (https://github.com/jOOQ/jOOQ/issues/15079), 3.17.14 (https://github.com/jOOQ/jOOQ/issues/15080), 3.16.20 (https://github.com/jOOQ/jOOQ/issues/15081)

**Comment by lukaseder** [comment]
https://github.com/jOOQ/jOOQ/issues/9906 has been implemented and a Maven repository for commercial artifacts is now available at https://repo.jooq.org
